syntax = "proto3";

package Blocksettle.Communication.MarketDataHistory;

enum Interval
{
	OneYear         = 0;
	SixMonths       = 1;
	OneMonth        = 2;
	OneWeek         = 3;
	TwentyFourHours = 4;
	TwelveHours     = 5;
	SixHours        = 6;
	OneHour         = 7;
};

enum MarketDataHistoryMessageType
{
	ProductsListType = 0;
	OhlcHistoryType  = 1;
};

message MarketDataHistoryRequest
{
	MarketDataHistoryMessageType request_type = 1;
	bytes                        request      = 2;
};

message MarketDataHistoryResponse
{
	MarketDataHistoryMessageType response_type = 1;
	bytes                        response      = 2;
};

message OhlcRequest
{
	string   product  = 1;
	Interval interval = 2;
    uint32   limit    = 3;
};

message OhlcCandle
{
	uint64 timestamp = 1;
	uint64 start     = 2;
	uint64 end       = 3;
	double open      = 4;
	double high      = 5;
	double low       = 6;
	double close     = 7;
	uint64 trades    = 8;
	double volume    = 9;
};

message OhlcResponse
{
    string              product  = 1;
    Interval            interval = 2;
    repeated OhlcCandle candles  = 3;
};

message ProductsListResponse
{
    repeated string products = 1;
};